### PR TITLE
v1.26 adoption

### LINKED ISSUE #2051: v1.26 release
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind bug
> /kind feature

**What happened**:
k8s 1.26 is out
we should consider create corresponding build 

**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- openstack-cloud-controller-manager(or other related binary) version:
- OpenStack version:
- Others:

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

**Which issue this PR fixes(if applicable)**:
fixes #2051

- update to k8s 1.26 as dependency
- fix k8s.io/apimachinery/pkg/util/sets string type deprecation lead to warning 
- other warnings related to code style
- ignore a manila test temply
- replace corev1.LoadBalancerStatus with networkingv1.IngressLoadBalancerStatus
- git version string lead to CI issue

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kms/server/server.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 	"golang.org/x/sys/unix"
 	"google.golang.org/grpc"
 	gcfg "gopkg.in/gcfg.v1"
-	pb "k8s.io/apiserver/pkg/storage/value/encrypt/envelope/v1beta1"
+	pb "k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/v2alpha1"
```
Comment (ramineni) [inline pkg/kms/server/server.go:12]: i guess the functions have been moved here , kmsapi "k8s.io/kms/apis/v1beta1" ?
https://github.com/kubernetes/apiserver/blob/master/pkg/storage/value/encrypt/envelope/grpc_service.go#L86

**Comment by jichenjc** [comment]
@gman0 looks like manila test keep failed due to 
after I upgrade to 1.26 (I didn't do direct update on the code logic of manila, only minor param change)
can you help provide some insight or we might want to skip this test for now and continue the 1.26 
and open issue to track?
```
Summarizing 1 Failure:
  [FAIL] [sig-storage] [manila-csi-e2e] CSI Volumes [Driver: nfs.manila.csi.openstack.org] [Testpattern: Dynamic PV (default fs)] provisioning [It] should provision storage with snapshot data source [Feature:VolumeSnapshotDataSource]
```

**Comment by jichenjc** [comment]
@ramineni @gman0 I updated the test to temply skip the failed manila case
we can open issue to track to add it back

**Comment by ramineni** [comment]
@jichenjc Im ok with merging ahead this PR. But the bug has to be tracked as release blocking ?, IIUC manila driver would have broken snapshot functionality if released without the fix?

**Comment by jichenjc** [comment]
>>> But the bug has to be tracked as release blocking 

agree, I will wait for a few days then merge this if we can't fix in short term and open an issue to track for release blocker

**Comment by gman0** [comment]
Sorry for the delay, I'll work on this today/tomorrow.

**Comment by ramineni** [comment]
@gman0 the test case is skipped right now , so you might not find the failed logs

**Comment by gman0** [comment]
Ah I didn't notice the push.

**Comment by jichenjc** [comment]
I think we can merge this and let's create a new issue to track/revert the skip? 
not sure the effort and time needed for this , thoughts?

**Comment by gman0** [comment]
I'm fine with both as long as the release is cut once we fix this.

**Comment by jichenjc** [comment]
@ramineni @gman0 can you help merge this? I will open another issue and assign to @gman0 to help

**Pull request merged**